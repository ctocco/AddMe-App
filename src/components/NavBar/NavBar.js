import React from "react";

const NavBar = () => {
  return (
    <div>
      <h1>"hello"</h1>
    </div>
  );
};

export default NavBar;
