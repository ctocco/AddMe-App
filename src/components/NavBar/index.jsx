import React from "react";
import { Link } from "@reach/router";
import styled from "styled-components";
import AdminPic from "../../assets/img/AdminPic.png";
import home from "../../assets/icons/home.svg";
import add_user from "../../assets/icons/add_user.svg";
import view_users from "../../assets/icons/view_users.svg";

const Nav = styled.nav`
  display: grid;
  background: #39555b;
  grid-template-columns: auto;
  grid-template-rows: 200px 200px 600px;
`;

const LogoContainer = styled.div`
  display: grid;
  grid-temlate-columns: 1fr;
  grid-template-rows: 150px auto;
  justify-content: center;
  text-align: center;
  align-content: center;
  color: white;
  grid-gap: 5px;
`;

const Circle = styled.div`
  display: flex;
  justify-content: center;
  background: #ddd;
  width: 150px;
  height: 150px;
  border-radius: 50%;
`;

const NavLink = styled.div`
  display: flex;
  color: white;
  align-items: center;
`;

const LinksContainer = styled.section`
  display: grid;
  grid-template-rows: 1fr 1fr 1fr;
  border-top: 1px solid white;
  padding: 20px;
  justify-content: center;
`;

const NavBar = () => {
  return (
    <Nav>
      <LogoContainer>
        <Circle>
          <img src={AdminPic} alt={"placeholder of a key"} />
        </Circle>
        <h1 data-testid="h1">Admin</h1>
      </LogoContainer>

      <LinksContainer>
        <Link to="/">
          <NavLink>
            <img src={home} alt={"icon of home button"} />
            <div>Home</div>
          </NavLink>
        </Link>
        <Link to="/viewusers">
          <NavLink>
            <img src={view_users} alt={"icon of view users button"} />
            <div>View Users</div>
          </NavLink>
        </Link>
        <Link to="/addusers">
          <NavLink>
            <img src={add_user} alt={"icon of add user button"} />
            <div>Add Users</div>
          </NavLink>
        </Link>
      </LinksContainer>

      <div>Rest</div>
    </Nav>
  );
};

export default NavBar;
