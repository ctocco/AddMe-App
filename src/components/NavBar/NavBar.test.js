import React from "react";
import "@testing-library/jest-dom/extend-expect";
import {
  render,
  fireEvent,
  waitForElement,
  cleanup
} from "@testing-library/react";
import NavBar from "./index";

afterEach(cleanup);

test("renders", () => {
  const { asFragment } = render(<NavBar text="Hello!" />);
  expect(asFragment()).toMatchSnapshot();
});

it("inserts text in H1", () => {
  const { getByTestId, getByText } = render(<NavBar text="Hello!" />);
});
