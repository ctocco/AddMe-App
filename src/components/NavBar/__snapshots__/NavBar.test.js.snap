// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <nav
    class="sc-bdVaJa kCcGsS"
  >
    <a
      aria-current="page"
      href="/"
    >
      <h3>
        Admin
      </h3>
    </a>
    | 
    <a
      href="/displayusers"
    >
      Display Users
    </a>
  </nav>
</DocumentFragment>
`;
